src/mondrian_forest/mondrian_tree_reg.rs

```diff
@@ -0,0 +1,684 @@
+                node.range_min.to_vec(),
+                node.range_max.to_vec(),
+                node.threshold,
+                feature,
+                node.sums,
+                node.counts,
+            )?;
+
+            let mut children = vec![];
+            if let Some(left) = node.left {
+                children.push((left, node.right.is_none())); // Left child
+            }
+            if let Some(right) = node.right {
+                children.push((right, true)); // Right child
+            }
+
+            for (i, (child, last)) in children.into_iter().enumerate() {
+                self.recursive_repr(Some(child), f, &(prefix.to_owned() + child_prefix), last)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+impl<F: FType> Regressor<F> for MondrianTreeRegressor<F> {
+    fn learn_one(&mut self, x: &Array1<F>, y: &RegTarget<F>) {
+        self.root = match self.root {
+            None => Some(self.create_leaf(x, y, None, F::zero())),
+            Some(root_idx) => Some(self.go_downwards(root_idx, x, y)),
+        };
+        // println!("learn_one() tree post {}===========", self);
+    }
+
+    fn predict_one(&mut self, x: &Array1<F>, y: &RegTarget<F>) -> F {
+        // self.test_tree();
+        self.predict(x, self.root.unwrap(), F::one())
+    }
+}
+
+impl<F: FType> MondrianTreeRegressor<F> {
+    pub fn new(n_features: usize) -> Self {
+        MondrianTreeRegressor::<F> {
+            n_features,
+            rng: rand::thread_rng(),
+            nodes: vec![],
+            root: None,
+        }
+    }
+
+    fn create_leaf(
+        &mut self,
+        x: &Array1<F>,
+        y: &RegTarget<F>,
+        parent: Option<usize>,
+        time: F,
+    ) -> usize {
+        let mut node = NodeRegressor::<F> {
+            parent,
+            time,
+            is_leaf: true,
+            range_min: x.clone(),
+            range_max: x.clone(),
+            feature: usize::MAX,
+            threshold: F::infinity(),
+            left: None,
+            right: None,
+            sums: F::zero(),
+            counts: 0,
+            n_features: self.n_features,
+        };
+        node.add(x, y);
+        self.nodes.push(node);
+        let node_idx = self.nodes.len() - 1;
+        node_idx
+    }
+
+    fn create_empty_node(&mut self, parent: Option<usize>, time: F) -> usize {
+        let node = NodeRegressor::<F> {
+            parent,
+            time,
+            is_leaf: true,
+            range_min: Array1::from_elem(self.n_features, F::infinity()),
+            range_max: Array1::from_elem(self.n_features, -F::infinity()),
+            feature: usize::MAX,
+            threshold: F::infinity(),
+            left: None,
+            right: None,
+            sums: F::zero(),
+            counts: 0,
+            n_features: self.n_features,
+        };
+        self.nodes.push(node);
+        let node_idx = self.nodes.len() - 1;
+        node_idx
+    }
+
+    fn test_tree(&self) {
```

Comment:
Do you think the test suite could also be shared by the classification and the regression trees>